cpp/include/cudf/detail/aggregation/aggregation.hpp

```diff
@@ -1219,6 +1261,18 @@ struct target_type_impl<Source, aggregation::MIN> {
   using type = Source;
 };
 
+// Computing MAX_BY of Source, use Source accumulator
+template <typename Source>
+struct target_type_impl<Source, aggregation::MAX_BY> {
+  using type = struct_view;
+};
+
+// Computing MIN_BY of Source, use Source accumulator
+template <typename Source>
+struct target_type_impl<Source, aggregation::MIN_BY> {
+  using type = struct_view;
```

Comment:
So the output type is the same type as the input? If so, I would recommend to add a parameter to the constructor of min_by/max_by aggregation to specify which column index to compute arg_min/arg_max. By doing so, if in the future we need to compute min_by/max_by of more than one column (i.e., get values of several columns based on the same min/max value of the order by column) then we will trivially have this supported.

Reply:
> So the output type is the same type as the input?

Yes.

The struct is packed manually in plugin when doing min/max_by, not the original schema of table. So we can just always make the column which need to be argmin/max as the last column without specifying the index.

Reply:
Oh, current code actually always selects the second column. Will make the ordering column be the first one and noted in comment.

Reply:
done.

Reply:
> the ordering column be the first one

Yes the first column as the ordering column makes more sense than the last one :+1: